### PR TITLE
Fix linking with python in macOS

### PR DESCRIPTION
**Pull request by tadeu**
See https://blog.tim-smith.us/2015/09/python-extension-modules-os-x/
and https://github.com/conda-forge/cgal-feedstock/pull/41

### CODE REVIEW

**Comment by tadeu** [comment]
This would benefit from #110

File: CMakeLists.txt [hunk 1]
```diff
@@ -12,7 +12,11 @@ if (POLICY CMP0042)
   cmake_policy (SET CMP0042 NEW)
 endif()
 
-option ( LINK_PYTHON_LIBRARY "Link against python libraries" ON )
+if ( APPLE )
+  option ( LINK_PYTHON_LIBRARY "Link against python libraries - NOT RECOMMENDED" OFF )
+else ()
+  option ( LINK_PYTHON_LIBRARY "Link against python libraries" ON )
+endif ()
```
Comment (lrineau) [inline CMakeLists.txt]: Is that change still required, when `-undefined dynamic_lookup` is added to link flags?

Comment (tadeu) [reply CMakeLists.txt]: I think so, since the recommendations are: "Please use -undefined dynamic_lookup *instead of* -lpython or -framework Python"

Comment (xantares) [reply CMakeLists.txt]: the LINK_PYTHON_LIBRARY option can be removed, then linking is only needed on windows, so the code would look like this:

if (WIN32)
  swig_link_libraries (${MODULENAME}_python ${PYTHON_LIBRARIES})
elseif(APPLE)
  set_target_properties(${MODULENAME}_python PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
endif ()

**Comment by xantares** [comment]
the LINK_PYTHON_LIBRARY option can be removed, then linking is only needed on windows, so the code would look like this:

if (WIN32)
  swig_link_libraries (${MODULENAME}_python ${PYTHON_LIBRARIES})
elseif(APPLE)
  set_target_properties(${MODULENAME}_python PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
endif ()

File: CMakeLists.txt (hunk 1 above)
Comment (lrineau) [reply CMakeLists.txt]: Oh, right. As one no longer links with libpython, there are undefined symbol, and that is why the `-undefined` flags is required. OK, thanks.

**Comment by tadeu** [comment]
I'm testing this here: https://github.com/conda-forge/cgal-feedstock/pull/48

It's still not working, `"-undefined dynamic_lookup"` is not being passed down to some targets, I'll review this later

File: cmake/Modules/SWIG_CGAL_Macros.cmake [hunk 2]
```diff
@@ -16,16 +16,19 @@ MACRO(ADD_SWIG_CGAL_LIBRARY libname)
   include_directories(${CMAKE_CURRENT_SOURCE_DIR})
   set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${COMMON_LIBRARIES_PATH}")
   set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${COMMON_LIBRARIES_PATH}")
-  EXTRACT_CPP_AND_LIB_FILES(${ARGN}) 
+  EXTRACT_CPP_AND_LIB_FILES(${ARGN})
   add_library(${libname} SHARED ${source_files})
   target_link_libraries(${libname} ${libstolinkwith})
+  if(APPLE)
+    set_target_properties(${libname} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+  endif()
```
Comment (xantares) [inline cmake/Modules/SWIG_CGAL_Macros.cmake]: it's only needed for swig modules, not these sublibs

Comment (tadeu) [reply cmake/Modules/SWIG_CGAL_Macros.cmake]: Yeah, I noticed that later. So, this is recommended to be added _strictly_ when linking with Python, right?

Comment (xantares) [reply cmake/Modules/SWIG_CGAL_Macros.cmake]: anything that's a plugin

**Comment by tadeu** [comment]
Just to let you know that the patch is good at https://github.com/conda-forge/cgal-feedstock/pull/48
